src/mixins/coordinate-grid-raster-mixin.js

```diff
@@ -1023,20 +1050,40 @@ export default function coordinateGridRasterMixin (_chart, _mapboxgl, browser) {
     _chart.prepareLockAxis("y")
   }
 
+  _chart._prepareY2Axis = function (g, y, transitionDuration) {
+    y.range([Math.round(_chart.yAxisHeight()), 0])
+
+    _y2Axis = _y2Axis.scale(y)
+
+    _y2Axis.ticks(_chart.effectiveHeight() / _y2Axis.scale().ticks().length < 16 ? Math.ceil(_chart.effectiveHeight() / 16) : 10)
+
+    _y2Axis.orient("right")
+
+    setYAxisFormat()
+
+    _chart._renderHorizontalGridLinesForAxis(g, y, _y2Axis, transitionDuration)
+    _chart.prepareLabelEdit("y2")
+    _chart.prepareLockAxis("y")
+  }
+
   _chart.renderYAxisLabel = function (axisClass, text, rotation, labelXPosition) {
     const root = _chart.root()
 
-    let yLabel = root.selectAll(".y-axis-label")
+    let yLabel = root.selectAll(`.${axisClass}-axis-label`)
 
     if (yLabel.empty()) {
       yLabel = root.append("div")
-        .attr("class", "y-axis-label")
+        .attr("class", `${axisClass}-axis-label`)
     }
 
     if (text !== "") {
       // TODO(croot): should add the rotation and labelXPosition here
       // As of now (09/02/2016) the chart.css is breaking this.
 
+      if (axisClass === "y2") {
```

Comment:
Not necessary but consts for these axis keys would be a good way to go IMO, as they're used/checked a few times throughout